src/paginators/base-paginator.js

```diff
@@ -0,0 +1,133 @@
+import cloneDeep from 'lodash.clonedeep';
+
+import * as actions from 'src/actions';
+import PaginatorError from 'src/paginator-error';
+
+
+export const errors = Object.freeze({
+  queryHandlerMissing: 'A query handler must be set to generate parameters'
+});
+
+export class BasePaginator {
+  constructor(request, requestOptions = null) {
+    this._cache = Object.create(null);
+    this._page = 0;
+    this._request = request;
+    this._requestOptions = cloneDeep(requestOptions);
+    this.queryHandler = null;
+  }
+
+  get page() {
```

Comment:
>ES6, exposing _page while not making it writable.

"it" in that sentence is referring to `_page` or `page`? I'm guessing `page`, since `_page` has an implicit getter and setter, the way you built it in the constructor